include/net/tls_credentials.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief TLS credentials management
+ *
+ * An API for applications to configure TLS credentials.
+ */
+
+#ifndef __TLS_CREDENTIAL_H
+#define __TLS_CREDENTIAL_H
+
+/**
+ * @brief TLS credentials management
+ * @defgroup tls_credentials TLS credentials management
+ * @ingroup networking
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** TLS credential types */
+enum tls_credential_type {
+	TLS_CREDENTIAL_NONE,
+	TLS_CREDENTIAL_CA_CERTIFICATE,
+	TLS_CREDENTIAL_PUBLIC_CERTIFICATE,
+	TLS_CREDENTIAL_PRIVATE_KEY,
+	TLS_CREDENTIAL_PSK,
+	TLS_CREDENTIAL_PSK_ID
+};
+
+/** Secure tag, a reference for TLS credential
+ *
+ * Secure tag can be used to reference credential after it was registered
+ * in the system.
+ *
+ * @note Some TLS credentials come in pairs:
+ *    - TLS_CREDENTIAL_PUBLIC_CERTIFICATE with TLS_CREDENTIAL_PRIVATE_KEY,
+ *    - TLS_CREDENTIAL_PSK with TLS_CREDENTIAL_PSK_ID.
+ *    Such pairs of credentials must have assigned the same secure tag to be
+ *    correctly handled in the system.
+ */
+typedef int sec_tag_t;
+
+/**
+ * @brief Add a TLS credential.
+ *
+ * @details This function adds a TLS credential, that can be used
+ *          by TLS/DTLS for authentication.
+ *
+ * @param tag     A security tag that credential will be referenced with.
+ * @param type    A TLS/DTLS credential type.
+ * @param cred    A TLS/DTLS credential.
+ * @param credlen A TLS/DTLS credential length.
+ *
+ * @return 0 if ok, < 0 if error.
+ */
+int tls_credential_add(sec_tag_t tag, enum tls_credential_type type,
+		       const void *cred, size_t credlen);
+
+/**
+ * @brief Get a TLS credential.
+ *
+ * @details This function gets an already registered TLS credential,
+ *          referenced by @p tag secure tag of @p type.
+ *
+ * @param tag     A security tag of requested credential.
+ * @param type    A TLS/DTLS credential type of requested credential.
+ * @param cred    A buffer for TLS/DTLS credential.
+ * @param credlen A buffer size on input. TLS/DTLS credential length on output.
+ *
+ * @return 0 if ok, < 0 if error.
```

Comment:
I said it in #7118, and think it's worth repeating here: I'd explicitly designated an error code for "access denied", to give users an expectation that credentials can be expected to be read back easily. (And yes, as it's security stuff, I think it's worth to do it right away.)